modules/ocf_filehost/manifests/init.pp

```diff
@@ -1,11 +1,43 @@
-class ocf_filehost {
+class ocf_filehost(
+  String $storage_device
+) {
   package { ['quotatool', 'nfs-kernel-server']:; }
 
-  file { '/etc/exports':
-    source  => 'puppet:///modules/ocf_filehost/exports',
-    require => Package['nfs-kernel-server'],
+  file {
+    '/etc/exports':
+      source  => 'puppet:///modules/ocf_filehost/exports',
+      require => Package['nfs-kernel-server'];
+
+    '/opt/homes':
+      ensure  => directory;
   }
 
+  mount {
+    '/opt/homes':
+      ensure  => mounted,
+      atboot  => true,
+      device  => $storage_device,
+      fstype  => 'ext4',
+      options => 'noacl,noatime,nodev,usrquota';
+
+    '/home':
```

Comment:
You might want to add an explicit dependency on `Mount['/opt/homes']` here. Otherwise you might find it bind-mounted the wrong directory.